xapian-core/common/wordaccess.h

```diff
@@ -27,6 +27,7 @@
 
 #include <cstdint>
 #include <type_traits>
+#include <cstring>
```

Comment:
This file does seem to use `memcpy()` without including the relevant header, but better not to make that unrelated fix in the same commit as changing the double serialisation.

Reply:
Separated into 2 commits.